modules/miner/update.go

```diff
@@ -193,12 +170,155 @@ func (m *Miner) deleteMapElementTxns(id splitSetID) {
 		m.blockMapHeap.removeSetByID(id)
 
 		// Promote sets from overflow heap to block if possible.
-		for overflowElem, canPromote := m.overflowMapHeap.peek(); canPromote && m.blockMapHeap.size+overflowElem.set.size < types.BlockSizeLimit-5e3; {
-			promotedElem := m.overflowMapHeap.pop()
-			m.blockMapHeap.push(promotedElem)
+		for overflowElem, canPromote := m.peekAtOverflow(); canPromote && m.blockMapHeap.size+overflowElem.set.size < types.BlockSizeLimit-5e3; {
+			promotedElem := m.popFromOverflow()
+			m.pushToBlock(promotedElem)
+		}
+		m.removeSplitSetFromUnsolvedBlock(id)
```

Comment:
Can you leave a comment explaining why you chose to remove the element from the block after you pushed elements into the block? Naively this seems backwards because you end up with an over-full block as a midstate, which could mean the block has to add more elements.

If there's a good reason to do it in this order, a comment is needed.